spring-cloud-gcp-samples/spring-cloud-gcp-vision-api-sample/src/main/java/com/example/VisionController.java

```diff
@@ -65,30 +59,15 @@
 	 */
 	@GetMapping("/vision")
 	public ModelAndView uploadImage(String imageUrl, ModelMap map) throws Exception {
-		// Copies the content of the image to memory.
-		byte[] imageBytes = StreamUtils.copyToByteArray(this.resourceLoader.getResource(imageUrl).getInputStream());
-
-		BatchAnnotateImagesResponse responses;
-
-		Image image = Image.newBuilder().setContent(ByteString.copyFrom(imageBytes)).build();
-
-		// Sets the type of request to label detection, to detect broad sets of categories in an image.
-		Feature feature = Feature.newBuilder().setType(Feature.Type.LABEL_DETECTION).build();
-		AnnotateImageRequest request =
-				AnnotateImageRequest.newBuilder().setImage(image).addFeatures(feature).build();
-		responses = this.imageAnnotatorClient.batchAnnotateImages(Collections.singletonList(request));
-
-		// We're only expecting one response.
-		if (responses.getResponsesCount() == 1) {
-			AnnotateImageResponse response = responses.getResponses(0);
+		AnnotateImageResponse response = cloudVisionTemplate.analyzeImage(
+				resourceLoader.getResource(imageUrl), Type.LABEL_DETECTION);
 
-			ImmutableMap.Builder<String, Float> annotations = ImmutableMap.builder();
-			for (EntityAnnotation annotation : response.getLabelAnnotationsList()) {
-				annotations.put(annotation.getDescription(), annotation.getScore());
-			}
-			map.addAttribute("annotations", annotations.build());
+		ImmutableMap.Builder<String, Float> annotations = ImmutableMap.builder();
+		for (EntityAnnotation annotation : response.getLabelAnnotationsList()) {
```

Comment:
You can process this with the Java `Stream` and produce `collect(Collectors.toMap())`